editoast/src/views/infra/mod.rs

```diff
@@ -591,17 +602,53 @@ pub mod tests {
             voltage: "0".into(),
             track_ranges: vec![],
         };
+        let rolling_stock: RollingStockModel = get_other_rolling_stock();
+        let rolling_stock_id = rolling_stock.id.unwrap();
+
         let req = create_object_request(infra.id.unwrap(), catenary.into());
         assert_eq!(call_service(&app, req).await.status(), StatusCode::OK);
+        let post_response = call_service(
+            &app,
+            TestRequest::post()
+                .uri("/rolling_stock")
+                .set_json(rolling_stock)
+                .to_request(),
+        )
+        .await;
+        assert_eq!(post_response.status(), StatusCode::OK);
 
         let req = TestRequest::get()
```

Comment:
To make the test easier to read, you can factorize the code (the test gets quite long otherwise) ;)
Proposal:
```let test_cases = [true, false];
for include_rolling_stock_modes in test_cases {
   //Checks which are common to both cases
   if include_rolling_stock_modes {
      //Assert modes are present: necessary checks for this case
   } else {
      //Assert modes are not present: necessary checks for this case
   }
}
```